builtin/game/item.lua

```diff
@@ -530,6 +530,18 @@ function core.node_dig(pos, node, digger)
 	end
 end
 
+function core.itemstring_with_palette(item, palette_index)
+	local stack = ItemStack(item) -- convert to ItemStack
```

Comment:
as i see you create temporary itemstacks to set a meta and retrieve the item string, it add many Lua overhead. Can you port those two functions coreside directly and generate the proper string ?